tests/utils/exec/cluster-autoscaler.go

```diff
@@ -0,0 +1,129 @@
+	h "github.com/terraform-redhat/terraform-provider-rhcs/tests/utils/helper"
+)
+
+type ClusterAutoscalerArgs struct {
+	Cluster                     string          `json:"cluster_id,omitempty"`
+	OCMENV                      string          `json:"ocm_environment,omitempty"`
+	BalanceSimilarNodeGroups    bool            `json:"balance_similar_node_groups,omitempty"`
+	SkipNodesWithLocalStorage   bool            `json:"skip_nodes_with_local_storage,omitempty"`
+	LogVerbosity                int             `json:"log_verbosity,omitempty"`
+	MaxPodGracePeriod           int             `json:"max_pod_grace_period,omitempty"`
+	PodPriorityThreshold        int             `json:"pod_priority_threshold,omitempty"`
+	IgnoreDaemonsetsUtilization bool            `json:"ignore_daemonsets_utilization,omitempty"`
+	MaxNodeProvisionTime        string          `json:"max_node_provision_time,omitempty"`
+	BalancingIgnoredLabels      []string        `json:"balancing_ignored_labels,omitempty"`
+	ResourceLimits              *ResourceLimits `json:"resource_limits,omitempty"`
+	ScaleDown                   *ScaleDown      `json:"scale_down,omitempty"`
+}
+type ResourceLimits struct {
+	Cores         *ResourceRange `json:"cores,omitempty"`
+	MaxNodesTotal int            `json:"max_nodes_total,omitempty"`
+	Memory        *ResourceRange `json:"memory,omitempty"`
+}
+type ScaleDown struct {
+	DelayAfterAdd        string `json:"delay_after_add,omitempty"`
+	DelayAfterDelete     string `json:"delay_after_delete,omitempty"`
+	DelayAfterFailure    string `json:"delay_after_failure,omitempty"`
+	UnneededTime         string `json:"unneeded_time,omitempty"`
+	UtilizationThreshold string `json:"utilization_threshold,omitempty"`
+	Enabled              bool   `json:"enabled,omitempty"`
+}
+type ResourceRange struct {
+	Max int `json:"max,omitempty"`
+	Min int `json:"min,omitempty"`
+}
+type ClusterAutoscalerService struct {
+	CreationArgs *ClusterAutoscalerArgs
+	ManifestDir  string
+	Context      context.Context
+}
+
+type ClusterAutoscalerOutput struct {
+	Cluster      string `json:"cluster_id,omitempty"`
+	LogVerbosity int    `json:"log_verbosity,omitempty"`
+}
+
+func (ca *ClusterAutoscalerService) Init(manifestDirs ...string) error {
+	ca.ManifestDir = CON.AWSVPCDir
+	if len(manifestDirs) != 0 {
+		ca.ManifestDir = manifestDirs[0]
+	}
+	ctx := context.TODO()
+	ca.Context = ctx
+	err := runTerraformInit(ctx, ca.ManifestDir)
+	if err != nil {
+		return err
+	}
+	return nil
+
+}
+
+func (ca *ClusterAutoscalerService) MagicImport(createArgs *ClusterAutoscalerArgs, extraArgs ...string) error {
+	ca.CreationArgs = createArgs
+	args, _ := combineStructArgs(createArgs, extraArgs...)
+	_, err := runTerraformApplyWithArgs(ca.Context, ca.ManifestDir, args)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (ca *ClusterAutoscalerService) Apply(createArgs *ClusterAutoscalerArgs, recordtfargs bool, extraArgs ...string) (string, error) {
```

Comment:
What's the differences between `MagicImport` and `Apply`? Both of them are using `terraform apply`